presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1248,13 +1259,31 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                 analysis.setColumn(field, columnHandle);
             }
 
-            if (isUpdateQuery) {
+            if (updateKind.isPresent()) {
                 // Add the row id field
-                ColumnHandle column = metadata.getUpdateRowIdColumnHandle(session, tableHandle.get());
-                Type type = metadata.getColumnMetadata(session, tableHandle.get(), column).getType();
+                ColumnHandle rowIdColumn;
```

Comment:
nit: s/rowIdColumn/rowIdColumnHandle/g to differentiate from just the rowId column

Reply:
Good point; changed.